crates/lib/src/command/supervisor.rs

```diff
@@ -367,3 +438,35 @@ async fn spawn_process(
 
 	Ok((proc, id))
 }
+
+/// Used to identify command registered with a Supervisor.
+#[must_use]
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Copy)]
+pub struct SupervisorId(NonZeroU64);
+
+impl SupervisorId {
+	/// Returns the id of the SupervisorId. This is guaranteed to not be 0.
```

Comment:
As it's meant to be an opaque type, let's _not_ have this.

Reply:
that is fair!

would you still want to drop the `id()` method all together, or make it private to the `crate`?

Reply:
just realized, `id()` seems to never be used internally, not sure why i added it at all hehe! that was quite silly! :D